landoapi/api/transplants.py

```diff
@@ -348,27 +322,39 @@ def post(data):
         author_name, author_email = select_diff_author(diff)
         timestamp = int(datetime.now().timestamp())
 
-        # Construct the patch that will be sent to transplant.
+        with db.session.begin_nested():
+            _lock_table_for(db.session, model=LandingJob)
+            lando_revision = Revision.query.filter(
+                Revision.revision_id == revision["id"],
+                Revision.diff_id == diff["id"],
+            ).one_or_none()
+            if not lando_revision:
+                # Create a new revision, but trigger an error on the landing job.
```

Comment:
This comment is out of date.

Reply:
Maybe raise an exception instead and wait for revision worker to pick these up instead.